concept/error.rs

```diff
@@ -49,7 +49,7 @@ typedb_error!(
     pub ConceptWriteError(component = "Concept write", prefix = "COW") {
         SnapshotGet(1, "Concept write failed due to a snapshot read error.", (source: SnapshotGetError)),
         SnapshotIterate(2, "Concept write failed due to a snapshot iteration error.", (source: Arc<SnapshotIteratorError>)),
-        ConceptRead(3, "Concept write failed due to a concept read error.", (source: ConceptReadError)),
+        ConceptRead(3, "Concept write failed due to a concept read error.", (source: Box<ConceptReadError>)),
```

Comment:
This can still be inlined, you just need to move out of the `Box` (`*` deref) in the `from()`

Reply:
I didn't quite understand what this meant, please feel free to improve it afterward in your PRs